<launch>
    
    <arg name="model" default="burger"/>
    <!-- <arg name="world" default="$(find turtlebot3_gazebo)/worlds/turtlebot3_world.world"/> -->
    <arg name="map_file" default="$(find turtlebot3_nav)/maps/map.yaml"/>
    <arg name="open_rviz" default="true"/>
    <arg name="cmd_vel_topic" default="/cmd_vel" />
    <arg name="odom_topic" default="/odom" />
    <arg name="move_forward_only" default="false"/>

    <!-- TurtleBot3 gazebo-->
    <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_world.launch">
        <arg name="model" value="$(arg model)"/>
        <!-- <arg name="world" value="$(arg world)"/> -->
    </include>

    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
        <arg name="model" value="$(arg model)" />
    </include>

    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

    <!-- AMCL localization -->
    <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>

    <!-- move_base for navigation -->
    <node pkg="move_base" name="move_base" type="move_base" output="screen">
        <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
        <rosparam file="$(find turtlebot3_nav)/param/costmap_common_params.yaml" command="load" ns="global_costmap"/>
        <rosparam file="$(find turtlebot3_nav)/param/costmap_common_params.yaml" command="load" ns="local_costmap"/>
        <rosparam file="$(find turtlebot3_nav)/param/dwa_local_planner_params.yaml" command="load"/>
        <rosparam file="$(find turtlebot3_nav)/param/global_costmap_params.yaml" command="load"/>
        <rosparam file="$(find turtlebot3_nav)/param/local_costmap_params.yaml" command="load"/>
        <rosparam file="$(find turtlebot3_nav)/param/move_base_params.yaml" command="load" />
        <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
        <remap from="odom" to="$(arg odom_topic)"/>
        <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    </node>

    <!-- rviz -->
    <group if="$(arg open_rviz)"> 
        <node pkg="rviz" name="rviz" type="rviz" required="true"
              args="-d $(find turtlebot3_nav)/rviz/turtlebot3_nav.rviz"/>
    </group>

    <!-- init_pose -->
    <node pkg="turtlebot3_nav" name="init_pose" type="init_pose.py" output="screen"/>

    <!-- goal_pose-->
    <node pkg="turtlebot3_nav" name="goal_pose" type="goal_pose.py" output="screen"/>
    
</launch>